src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -127,119 +126,235 @@ private function getPath(string $resourceShortName, array $operation, bool $coll
-                    'in' => 'body',
-                    'description' => sprintf('The updated %s resource', $resourceShortName),
-                    'schema' => ['$ref' => sprintf('#/definitions/%s', $resourceShortName)],
-                ],
-            ];
-            $pathOperation['responses'] ?? $pathOperation['responses'] = [
-                '200' => [
-                    'description' => sprintf('%s resource updated', $resourceShortName),
-                    'schema' => ['$ref' => sprintf('#/definitions/%s', $resourceShortName)],
-                ],
-                '400' => ['description' => 'Invalid input'],
-                '404' => ['description' => 'Resource not found'],
-            ];
+        if ('PUT' === $method) {
+            return $this->getPathOperationByMethodPut($pathOperation, $mimeTypes, $collection, $resourceMetadata, $resourceClass, $resourceShortName, $operationName, $definitions);
+        }
 
-            return $pathOperation;
+        if ('DELETE' === $method) {
+            return $this->getPathOperationByMethodDelete($pathOperation, $resourceShortName);
         }
+    }
 
-        if ($method === 'DELETE') {
-            $pathOperation['summary'] ?? $pathOperation['summary'] = sprintf('Removes the %s resource.', $resourceShortName);
-            $pathOperation['responses'] ?? $pathOperation['responses'] = [
-                '204' => ['description' => sprintf('%s resource deleted', $resourceShortName)],
-                '404' => ['description' => 'Resource not found'],
-            ];
+    /**
+     * @param \ArrayObject $pathOperation
+     * @param string       $resourceShortName
+     *
+     * @return mixed
+     */
+    private function getPathOperationByMethodDelete(\ArrayObject $pathOperation, string $resourceShortName) : \ArrayObject
+    {
+        $pathOperation['summary'] ?? $pathOperation['summary'] = sprintf('Removes the %s resource.', $resourceShortName);
+        $pathOperation['responses'] ?? $pathOperation['responses'] = [
+            '204' => ['description' => sprintf('%s resource deleted', $resourceShortName)],
+            '404' => ['description' => 'Resource not found'],
+        ];
+
+        $pathOperation['parameters'] ?? $pathOperation['parameters'] = [[
+            'name' => 'id',
+            'in' => 'path',
+            'type' => 'integer',
+            'required' => true,
+        ]];
+
+        return $pathOperation;
+    }
 
-            $pathOperation['parameters'] ?? $pathOperation['parameters'] = [[
+    /**
+     * @param \ArrayObject     $definitions
+     * @param bool             $collection
+     * @param ResourceMetadata $resourceMetadata
+     * @param string           $resourceClass
+     * @param string           $operationName
+     *
+     * @return string
+     */
+    private function getDefinitionResponseSchema(\ArrayObject &$definitions, bool $collection, ResourceMetadata $resourceMetadata, string $resourceClass, string $operationName) : string
```

Comment:
`getDefinitionResponseSchema` and `getDefinitionParameterSchema` are almost the same. Can't they be factorized?